### PR TITLE
Include <array> in pose_graph/node.h and pose_2d.h

### PR DESCRIPTION
**Pull request by ojura**
Makes MSVC happy.

### CODE REVIEW

**Comment by pifon2a** [comment]
Thank you, Juraj! I'll add it.

**Comment by pifon2a** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user pifon2a. Merge queue now has a length of 1.

File: cartographer/pose_graph/node.h [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 #include "cartographer/pose_graph/proto/node.pb.h"
 
+#include <array>
```
Comment (pifon2a) [inline cartographer/pose_graph/node.h]: it is probably not needed here. Only in derived classes.

**Comment by wally-the-cartographer** [comment]
Error: PUT https://api.github.com/repos/googlecartographer/cartographer/pulls/1280/merge: 405 At least 1 approving review is required by reviewers with write access. []

Log:
~~~
=> handle work item: PR 1280 on googlecartographer/cartographer.
 => Switching to branch master.
 $ git reset --hard HEAD [data/cartographer]
HEAD is now at 07a9efee Add metrics: real time ratio and cpu time ratio. (#1275)
 $ git clean -fd [data/cartographer]
 $ git checkout master [data/cartographer]
Already on 'master'
Your branch is up to date with 'origin/master'.
 $ git pull [data/cartographer]
Already up to date.
Current branch master is up to date.
 => Checking out ojura/patch-1 as local branch pr_1280.
 $ git remote add ojura git@github.com:ojura/cartographer.git [data/cartographer]
fatal: remote ojura already exists.
 $ git fetch ojura [data/cartographer]
 $ git branch --track pr_1280 ojura/patch-1 [data/cartographer]
Branch 'pr_1280' set up to track remote branch 'patch-1' from 'ojura'.
 => Switching to branch pr_1280.
 $ git reset --hard HEAD [data/cartographer]
HEAD is now at 07a9efee Add metrics: real time ratio and cpu time ratio. (#1275)
 $ git clean -fd [data/cartographer]
 $ git checkout pr_1280 [data/cartographer]
Switched to branch 'pr_1280'
Your branch is up to date with 'ojura/patch-1'.
 $ git pull --force [data/cartographer]
Already up to date.
Current branch pr_1280 is up to date.
 => merge master.
 $ git merge master [data/cartographer]
Already up to date.
 => Running clang-format on all *.cc and *.h files
 $ git commit -am Ran clang-format. [data/cartographer]
On branch pr_1280
Your branch is up to date with 'ojura/patch-1'.

nothing to commit, working tree clean
 $ git push --force ojura HEAD:patch-1 [data/cartographer]
Everything up-to-date
 => Abandoning branch pr_1280.
 => Switching to branch master.
 $ git reset --hard HEAD [data/cartographer]
HEAD is now at d5891251 Update pose_2d.h
 $ git clean -fd [data/cartographer]
 $ git checkout master [data/cartographer]
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
 $ git branch -D pr_1280 [data/cartographer]
Deleted branch pr_1280 (was d5891251).

~~~

File: cartographer/pose_graph/node.h (hunk 1 above)
Comment (ojura) [reply cartographer/pose_graph/node.h]: `AddParameterBlock` has `std::array` in its prototype, right?

**Pull request merged**